Adafruit_SPIDevice.cpp

```diff
@@ -112,7 +112,17 @@ bool Adafruit_SPIDevice::begin(void) {
 void Adafruit_SPIDevice::transfer(uint8_t *buffer, size_t len) {
   if (_spi) {
     // hardware SPI is easy
+
+#ifdef SPARK
+    // Spark uses asynchronous DMA for multi-byte transfers; 
+    // just use the single-byte method for sync transfers.
+    for (size_t i = 0; i < len; i++) {
+      _spi->transfer(buffer[i]);
+    }
+#else
```

Comment:
thats fine, please add whatever particle uses to 'wait' after doing the single transition instead of sending one byte at a time :)

Reply:
Reading the docs in more detail suggests I can just pass in `NULL` and get the DMA to be synchronous. I wasn't able to find any documentation on whether I can use the same buffer as the tx and rx buffer so I'm hoping this just works (...). If @zsup can get a Particle person to check this part out as well, that'd be awesome.

Reply:
@ladyada Let me know if my updates address all your concerns.